docs/src/components/home/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,279 @@
+
+const colorKeys = [10, 20, 40, 60, 80, 90, 100];
+const scale = ['primary', 'secondary', 'tertiary', 'success', 'info', 'error'];
+
+const Swatch = ({ color }) => (
+  <View height="2rem" flex="1" backgroundColor={color} borderRadius="medium" />
+);
+
+const Preview = ({ platform }) => {
+  const [exclusiveValue, setExclusiveValue] = React.useState('align-left');
+
+  const isMobile = useBreakpointValue({
+    base: true,
+    medium: false,
+  });
+
+  if (platform !== 'react') {
+    return (
+      <Authenticator
+        socialProviders={['amazon', 'apple', 'facebook', 'google']}
+      />
+    );
+  }
+
+  return (
+    <Grid
+      columnGap="small"
+      rowGap="small"
+      templateColumns={isMobile ? '1fr' : '1fr 1fr 1fr 1fr'}
+    >
+      <Card variation="elevated" columnStart="1" columnEnd="4">
+        <Flex direction="column">
+          <Alert variation="info" heading="Flash sale!" />
+          <Flex direction={isMobile ? 'column' : 'row'}>
+            <Image alt="" src="/amplify-placeholder.svg" width="100px" />
+            <Flex direction="column">
+              <ToggleButtonGroup
+                value={exclusiveValue}
+                size="small"
+                isExclusive
+                onChange={(value: string) => setExclusiveValue(value)}
+              >
+                <ToggleButton value="align-left" ariaLabel="align left">
+                  Sm
+                </ToggleButton>
+                <ToggleButton value="align-center" ariaLabel="align center">
+                  Md
+                </ToggleButton>
+                <ToggleButton value="align-right" ariaLabel="align right">
+                  Lg
+                </ToggleButton>
+                <ToggleButton value="align-justify" ariaLabel="align justify">
+                  XL
+                </ToggleButton>
+              </ToggleButtonGroup>
+              <Flex direction="row" alignItems="center">
+                <Text as="span" fontSize="xl">
+                  $29.99
+                </Text>
+                <Text as="span" color="font.success">
+                  20% off
+                </Text>
+              </Flex>
+              <Button variation="primary">Add to cart</Button>
+            </Flex>
+          </Flex>
+        </Flex>
+      </Card>
+      {isMobile ? null : (
+        <>
+          <Card variation="elevated" columnStart="4" columnEnd="-1">
+            <Flex direction="column">
+              <CheckboxField
+                label="Sprinkles"
+                value="Sprinkles"
+                name="topping"
+                defaultChecked
+              />
+              <CheckboxField label="Frosting" value="Frosting" name="topping" />
+              <RadioGroupField
+                label="Language"
+                name="language"
+                defaultValue="css"
+                labelHidden
+              >
+                <Radio value="html">html</Radio>
```

Comment:
I have to double-click on the Radio buttons to get them to fill in. Not a big deal, but this could be solved by making it a controlled component.

Reply:
Just FYI, the fix didn't have to do with controlled or uncontrolled, but because the theme and primitives section both have a RadioGroup with the same name so the browser thought they were part of the same input.